tool/bump_version.dart

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:io';
+
+void main() async {
+  await bumpVersionInAllPubspecFilesAndAddBlankEntryToChangelogs();
+}
+
+/// When executed, first prompts the user to enter old version, new version, and date.
+/// It then looks for instances of old version in pubspec files and updates them to new version.
+/// It also adds a blank entry to the top of CHANGELOG.md files.
+Future<void> bumpVersionInAllPubspecFilesAndAddBlankEntryToChangelogs() async {
+  Future<void> _addTextToTopOfFile(String filePath, String textToAdd) async {
+    final file = File(filePath);
+    var contents = await file.readAsString();
+    contents = '$textToAdd$contents';
+    await file.writeAsString(contents);
+  }
+
+  stdout.write('What is the old version number? \n');
+  String? oldVersion = stdin.readLineSync();
+
+  stdout.write('What is the new version number? \n');
+  String? newVersion = stdin.readLineSync();
+
+  stdout.write(
```

Comment:
Super minor concern over lack of validation for the user prompts but nothing it does is breaking and typos can just be fixed afterwards but it seems like an improvement we might consider at some point.

Reply:
I agree that would be a place to improve. It could also populate some of the input with a suggestion, like current date, or find old version from somewhere in the code. There are other CLI niceties that could be included, like outputting a summary of what was changed, but I thought this was good enough to start.